open-apparel-registry/open-apparel-registry
# Configure hot reloading for React app during development

## Overview

This PR configures hot reloading for the React app during development. To accomplish this without ejecting from CRA, I brought in

- React App Rewired (https://github.com/timarney/react-app-rewired) which enables customizing CRA webpack configs without ejecting
- React App Rewire Hot Loader (https://github.com/cdharris/react-app-rewire-hot-loader) which enables setting up hot reloading with react-app-rewired

Setting this up also required adjusting our development proxy settings because the previous setup would route all requests not caught by React Router to Django -- including the websocket requests used by Webpack Dev Server to push hot updates. (This was the source, btw, of the frequent error messages about the socket disconnecting during development). To fix that I followed the guide here -- https://facebook.github.io/create-react-app/docs/proxying-api-requests-in-development#configuring-the-proxy-manually -- to set up a manual proxy.

Since not all of our API routes are prefixed with `/api/`, I created a list of the routes from urls.py and then looped over them to set up proxies. It seemed like this change would have a smaller impact than trying to update the Django routing here. (I think we should eventually update the Django routing, but it's out of scope for this issue.)

The result is that we now have hot reloading in development.

Connects #549

## Notes

Second commit here drops an unused dependency on react-mapgl that I noticed when messing with the package.json file.

## Testing Instructions

- get branch, then `./scripts/update`
- once update has finished, visit the app at http://localhost:8081 and verify that it works as it did before
- `./scripts/server` then visit the development app at http://localhost:6543
- make some changes to the `FilterSidebarSearchTab` component locally, then save. Verify that after saving the changes are rendered without refreshing the page and that the previous app state is maintained

## Checklist
